core/lib/dal/src/blocks_dal.rs

```diff
@@ -423,16 +424,30 @@ impl BlocksDal<'_, '_> {
                 .with_arg("eth_tx_id", &eth_tx_id)
                 .execute(self.storage)
                 .await?;
+
+                if result.rows_affected() == 0 {
+                    let err = Instrumented::new("set_eth_tx_id#commit")
+                        .with_arg("number_range", &number_range)
+                        .with_arg("eth_tx_id", &eth_tx_id)
+                        .arg_error(
```

Comment:
Nit: This is a constraint error rather an arg one (an arg error should be possible to produce without running DB queries).